### PR TITLE
Hot fix faq invalid vc (EXPOSUREAPP-7709)

### PR DESCRIPTION
**Pull request by chiljamgossow**
device-2021-06-09-173604

### CODE REVIEW

**Review by NataliaLemmerth — DISMISSED** [review]
UA approved

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/ui/ErrorDialog.kt [hunk 1]
```diff
@@ -60,3 +60,14 @@ fun Throwable.toErrorDialogBuilder(context: Context) =
...
+fun Throwable.toErrorDialogBuilderNoDetails(context: Context) =
+    MaterialAlertDialogBuilder(context).apply {
+        val error = this@toErrorDialogBuilderNoDetails
+        val humanReadable = error.tryHumanReadableError(context)
+
+        setTitle(humanReadable.title ?: context.getString(R.string.errors_generic_headline_short))
+        setMessageView(humanReadable.description, textHasLinks = true)
+
+        setPositiveButton(R.string.errors_generic_button_positive) { _, _ -> }
+    }
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/ui/ErrorDialog.kt:73]: Does this require a new extension function?

If we use `toErrorDialogBuilder` and then call `setNeutralButton` on the builder, we overwrite the previous neutral?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/core/certificate/InvalidHealthCertificateException.kt [hunk 2]
```diff
@@ -48,20 +48,26 @@ class InvalidHealthCertificateException(
...
-            HC_ZLIB_DECOMPRESSION_FAILED,
-            HC_COSE_TAG_INVALID,
-            VC_PREFIX_INVALID,
-            VC_HC_CWT_NO_DGC,
-            VC_HC_CWT_NO_EXP,
-            VC_HC_CWT_NO_HCERT,
-            VC_HC_CWT_NO_ISS,
-            VC_JSON_SCHEMA_INVALID,
-            -> CachedString { context ->
+            in codesVcInvalid -> CachedString { context ->
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/core/certificate/InvalidHealthCertificateException.kt:70]: Nice :heart_eyes:

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/scan/VaccinationQrCodeScanViewModel.kt [hunk 3]
```diff
@@ -21,14 +22,19 @@ class VaccinationQrCodeScanViewModel @AssistedInject constructor(
...
+
     fun onScanResult(barcodeResult: BarcodeResult) = launch {
         try {
             event.postValue(Event.QrCodeScanInProgress)
             val qrCode = vaccinationQRCodeValidator.validate(barcodeResult.text)
             val vaccinationCertificate = vaccinationRepository.registerVaccination(qrCode)
             event.postValue(Event.QrCodeScanSucceeded(vaccinationCertificate.personIdentifier.codeSHA256))
         } catch (e: Throwable) {
-            errorEvent.postValue(e)
+            if (e is InvalidHealthCertificateException && e.showFaqButton) {
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/scan/VaccinationQrCodeScanViewModel.kt:34]: Should we just check that before creating the dialog, and then only use the `errorEvent`, without the additional `errorEventWithFaq`?

**Review by AlexanderAlferov — DISMISSED** [review]

**Review by NataliaLemmerth — APPROVED** [review]
UA approved

**Review by d4rken — APPROVED** [review]

**Review by AlexanderAlferov — APPROVED** [review]

**Pull request merged**